packages/kubevirt/charts/templates/namespace-hooks.yaml

```diff
@@ -0,0 +1,71 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  namespace: {{ .Release.Namespace }}
+  name: kubevirt-namespace-modifier
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": hook-succeeded
+    "helm.sh/hook-weight": "1"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: kubevirt-namespace-modifier
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": hook-succeeded
+    "helm.sh/hook-weight": "1"
+rules:
+  - apiGroups: [ "" ]
+    resources: [ "namespaces" ]
+    resourceNames:
+      - {{ .Release.Namespace | quote }}
+    verbs: [ "get", "patch" ]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: kubevirt-namespace-modifier
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": hook-succeeded
+    "helm.sh/hook-weight": "2"
+subjects:
+  - kind: ServiceAccount
+    namespace: {{ .Release.Namespace }}
+    name: kubevirt-namespace-modifier
+roleRef:
+  kind: Role
+  name: kubevirt-namespace-modifier
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  namespace: {{ .Release.Namespace }}
+  name: kubevirt-namespace-modifier
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": hook-succeeded
+    "helm.sh/hook-weight": "3"
+spec:
+  template:
+    metadata:
+      name: kubevirt-namespace-modifier
+    spec:
+      serviceAccountName: kubevirt-namespace-modifier
+      restartPolicy: OnFailure
+      containers:
+        - name: kubevirt-namespace-modifier
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: {{ .Values.kubecli.image }}
+          command:
+            - /bin/sh
+            - -c
+            - >
+              kubectl label namespace {{ .Release.Namespace }} kubevirt.io="";
+              kubectl label namespace {{ .Release.Namespace }} pod-security.kubernetes.io/enforce="privileged";
```

Comment:
These are labels which the upstream manifests are shipped with. Unfortunately, there wasn't an easier way to do this than a pre-install hook... https://github.com/helm/helm/issues/3503#issuecomment-696712443

Reply:
My only concern is that everything related to this is a leftover that will live in the cluster for no reason. Honestly I don't know how to _fix_ it but maybe the container can remove "everything" after the namespace is labeled? I mean, running kubectl delete role kubevirt-namespace-modifier, etc.

Reply:
They won't live forever, the delete policy drops them all immediately.

```
root@atanas:~/charts/charts# helm install kubevirt kubevirt/0.2.0 -n kubevirt --create-namespace --debug
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /root/charts/charts/kubevirt/0.2.0

client.go:128: [debug] creating 1 resource(s)
install.go:151: [debug] CRD kubevirts.kubevirt.io is already present. Skipping.
client.go:128: [debug] creating 1 resource(s)
client.go:128: [debug] creating 1 resource(s)
client.go:128: [debug] creating 1 resource(s)
client.go:128: [debug] creating 1 resource(s)
client.go:128: [debug] creating 1 resource(s)
client.go:540: [debug] Watching for changes to Job kubevirt-namespace-modifier with timeout of 5m0s
client.go:568: [debug] Add/Modify event for kubevirt-namespace-modifier: ADDED
client.go:607: [debug] kubevirt-namespace-modifier: Jobs active: 0, jobs failed: 0, jobs succeeded: 0
client.go:568: [debug] Add/Modify event for kubevirt-namespace-modifier: MODIFIED
client.go:607: [debug] kubevirt-namespace-modifier: Jobs active: 1, jobs failed: 0, jobs succeeded: 0
client.go:568: [debug] Add/Modify event for kubevirt-namespace-modifier: MODIFIED
client.go:607: [debug] kubevirt-namespace-modifier: Jobs active: 0, jobs failed: 0, jobs succeeded: 0
client.go:568: [debug] Add/Modify event for kubevirt-namespace-modifier: MODIFIED
client.go:310: [debug] Starting delete for "kubevirt-namespace-modifier" ServiceAccount
client.go:310: [debug] Starting delete for "kubevirt-namespace-modifier" Role
client.go:310: [debug] Starting delete for "kubevirt-namespace-modifier" RoleBinding
client.go:310: [debug] Starting delete for "kubevirt-namespace-modifier" Job
```

Reply:
You can ignore my comment then :D Sorry for the noise.